File: Solutions/4068/java/program.java
// Copy from Quera

import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt(), k = scanner.nextInt();
        int [] arr = new int[n+1];
        arr[1] = 1;
        for(int i = 2; i<= n; i++){
            if(i == n && arr[i-1] + k <= n ){
                arr[i] = arr[i-1] + k;
                break;
            }
            if(i%2 == 0){
                if(arr[i-1]+(k + 1) > n){
                    System.out.println("Impossible");
                    return;
                }
                arr[i] = arr[i-1] + (k + 1);
            }else if(i % 2 == 1){
                arr[i] = arr[i-1] - k;
            }
        }
        for(int i = 1; i <=n; i++)System.out.print(arr[i] + " ");
    }
}

File: Solutions/4068/c++/program.cpp
// Copy from Quera

#include <iostream>
using namespace std;

int main ()
{
	long long n, k;
	cin >> n >> k;
	if ( k >n / 2 )
	{
		cout << "Impossible" << endl;
	}
	else
	{
	    if(n % 2 == 0 || n % 2 == 1)
	    {
		    for(int i = (n / 2); i > 0; i --)
		    {
			    cout << i << " " << i + (n / 2) << " ";
		    }
		    cout << n;
	    }
	}
}
